go-controller/pkg/ovn/egressgw.go

```diff
@@ -208,7 +209,13 @@ func (oc *DefaultNetworkController) isPodInLocalZone(podName ktypes.NamespacedNa
 
 	pod, err := oc.watchFactory.PodCoreInformer().Lister().Pods(podName.Namespace).Get(podName.Name)
 	if err != nil {
-		return false, err
+		if !apierrors.IsNotFound(err) {
```

Comment:
we dont want to do this kind of logic here because this assumes it is OK for the pod to be missing all of the time. In the add case we want to propagate an error up the stack and let the handler deal with it, because we were asked to add this thing and now we cant find it. It isn't up to this function to decide.